packages/core/src/application_ref.ts

```diff
@@ -764,7 +761,7 @@ export class ApplicationRef {
       for (let view of this._views) {
         view.detectChanges();
       }
-      if (this._enforceNoNewChanges) {
+      if (typeof ngDevMode === 'undefined' || ngDevMode) {
```

Comment:
I think this one might be a bit tricky... For the cases when `ngDevMode` is not declared (i.e. for projects that use ViewEngine and don't use CLI), this code would not be tree-shaken and would be invoked in prod mode. This would be a change in behavior since currently it's not executed due to the `isDevMode()`-based check. I believe we may still achieve tree-shaking by just appending the `ngDevMode` to the existing check like this:
```suggestion
      if ((typeof ngDevMode === 'undefined' || ngDevMode) && isDevMode()) {
        for (let view of this._views) {
          view.checkNoChanges();
        }
      }
```
It'd be also helpful to leave a comment above the `if` statement to provide more context for future refactoring (when ViewEngine code is removed).